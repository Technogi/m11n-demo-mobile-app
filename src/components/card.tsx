import React from 'react'
import { View, StyleSheet, ViewStyle } from 'react-native'
import { moderateScale } from 'react-native-size-matters'

/**
 * Returns a card component.
 *
 * @param {({
 *   children?: JSX.Element | JSX.Element[]
 *   style?: ViewStyle
 * })} {
 *   children: child component,
 *   style: view style
 * }
 * @returns {JSX.Element} JSX.Element
 */
const Card = ({ children, style }: { children?: JSX.Element | JSX.Element[]; style?: ViewStyle }): JSX.Element => {
  const { viewStyle } = styles

  return <View style={{ ...viewStyle, ...style }}>{children}</View>
}

const styles = StyleSheet.create({
  viewStyle: {
    backgroundColor: 'white',
    width: '100%',
    minHeight: 60,
    borderRadius: moderateScale(10),
    padding: moderateScale(15),
  },
})

export default Card
