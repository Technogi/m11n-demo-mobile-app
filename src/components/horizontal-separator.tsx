import React from 'react'
import { View } from 'react-native'

/**
 * Returns a horizontal separator that can be used to give space between components.
 *
 * @param {{ space: number }} { space }
 * @return {JSX.Element}  {JSX.Element}
 */
const HorizontalSeparator = ({ space }: { space: number }): JSX.Element => (
  <View style={{ borderBottomWidth: 0.5, borderBottomColor: '#939393', marginTop: space }} />
)

export default HorizontalSeparator
