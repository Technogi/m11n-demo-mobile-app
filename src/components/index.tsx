import Container from './layout/container'
import Content from './layout/content'
import Card from './card'
import PressableOpacity from './buttons/pressable-opacity'
import Button from './buttons/button'
import toastConfig from './toast/toast-config'
import Headline from './text/headline'
import Subtitle from './text/subtitle'
import Body from './text/body'
import Caption from './text/caption'
import CustomTextInput from './forms/text-input/custom-text-input'
import TextInputForm from './forms/text-input/text-input.form'
import SecondaryButton from './buttons/secondary.button'
import LogoHeader from './logo-header'
import TitleHeader from './title-header'

import { ToastType } from './toast/types'

export {
  Container,
  Content,
  Card,
  PressableOpacity,
  Button,
  SecondaryButton,
  toastConfig,
  ToastType,
  Headline,
  Subtitle,
  Body,
  Caption,
  CustomTextInput,
  TextInputForm,
  LogoHeader,
  TitleHeader,
}
