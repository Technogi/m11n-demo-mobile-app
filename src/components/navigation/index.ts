import MainTabIcon from './main-tab-icon'

export { MainTabIcon }
