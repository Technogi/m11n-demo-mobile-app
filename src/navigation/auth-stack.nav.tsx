import React from 'react'
import { createNativeStackNavigator } from '@react-navigation/native-stack'

import screens from 'src/screens'

import { AuthNavName, AuthStackParamList } from './types'

const AuthStack = createNativeStackNavigator<AuthStackParamList>()

/**
 * Auth Stack Navigator: login and sign up screens
 *
 * @returns {JSX.Element} JSX.Element
 */
const AuthStackScreen = (): JSX.Element => (
  <AuthStack.Navigator initialRouteName={AuthNavName.LOGIN}>
    <AuthStack.Screen name={AuthNavName.LOGIN} component={screens.LoginScreen} options={{ headerShown: false }} />
  </AuthStack.Navigator>
)

export default AuthStackScreen
