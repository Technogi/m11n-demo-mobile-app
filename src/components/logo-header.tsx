import { StyleSheet, View, ViewStyle } from 'react-native'
import React from 'react'
import { moderateScale } from 'react-native-size-matters'
import MaterialIcons from 'react-native-vector-icons/MaterialIcons'

import { theme } from 'src/styles'

import Headline from './text/headline'

const LogoHeader = ({ containerStyle }: { containerStyle?: ViewStyle }): JSX.Element => {
  const { titleStyle, headlineStyle } = styles

  return (
    <View style={{ ...titleStyle, ...containerStyle }}>
      <MaterialIcons name="analytics" color={theme.PRIMARY_COLOR} size={theme.ICON_SIZE_7XL} />
      <Headline style={headlineStyle}>POStats</Headline>
    </View>
  )
}

const styles = StyleSheet.create({
  titleStyle: {
    marginVertical: moderateScale(20),
    flexDirection: 'row',
    alignItems: 'center',
  },
  headlineStyle: {
    fontSize: theme.FONT_SIZE_3XL,
    marginLeft: moderateScale(10),
  },
})

export default LogoHeader
