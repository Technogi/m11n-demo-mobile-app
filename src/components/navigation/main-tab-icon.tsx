/* eslint-disable @typescript-eslint/no-unused-vars */
import { View } from 'react-native'
import React from 'react'
import { moderateScale } from 'react-native-size-matters'
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons'

import { mainStyle } from 'src/styles'

/**
 * Returns the icon used for tabs.
 *
 * @param {{
 *   iconName: string
 *   focused: boolean
 *   color: string
 *   size: number
 * }} {
 *   iconName,
 *   focused,
 *   color,
 *   size,
 * }
 * @return {JSX.Element}  {JSX.Element}
 */
const MainTabIcon = ({ iconName, color, size }: { iconName: string; color: string; size: number }): JSX.Element => (
  <View
    style={{
      ...mainStyle.circleStyle(35),
      justifyContent: 'center',
      alignItems: 'center',
    }}
  >
    <View style={{ marginLeft: moderateScale(1), marginBottom: moderateScale(1) }}>
      <View style={{ marginLeft: moderateScale(1), marginBottom: moderateScale(1) }}>
        <MaterialCommunityIcons name={iconName} color={color} size={size} />
      </View>
    </View>
  </View>
)

export default MainTabIcon
