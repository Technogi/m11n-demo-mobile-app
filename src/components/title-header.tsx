import { StyleSheet, View, ViewStyle } from 'react-native'
import React from 'react'
import { moderateScale } from 'react-native-size-matters'
import Feather from 'react-native-vector-icons/Feather'
import { NativeStackNavigationProp } from '@react-navigation/native-stack'

import { theme } from 'src/styles'

import Headline from './text/headline'
import PressableOpacity from './buttons/pressable-opacity'

const TitleHeader = ({
  title,
  containerStyle,
  navigation,
}: {
  title: string
  containerStyle?: ViewStyle
  navigation: NativeStackNavigationProp<any, any>
}): JSX.Element => {
  const { titleStyle, headlineStyle } = styles

  return (
    <View style={{ ...titleStyle, ...containerStyle }}>
      <PressableOpacity hitSlop={5} containerStyle={{ flex: 0.1 }} onPress={() => navigation.goBack()}>
        <Feather name="chevron-left" color={theme.PRIMARY_COLOR} size={theme.ICON_SIZE_2XL} />
      </PressableOpacity>

      <View style={{ flex: 0.9 }}>
        <Headline style={headlineStyle}>{title}</Headline>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  titleStyle: {
    marginVertical: moderateScale(20),
    flexDirection: 'row',
    alignItems: 'center',
  },
  headlineStyle: {
    marginLeft: moderateScale(10),
  },
})

export default TitleHeader
