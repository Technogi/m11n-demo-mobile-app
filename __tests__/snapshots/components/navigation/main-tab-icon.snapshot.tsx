import React from 'react'
import renderer from 'react-test-renderer'
import { MainTabIcon } from 'src/components/navigation'

it('MainTabIcon renders correctly', () => {
  const tree = renderer.create(<MainTabIcon iconName="home" color="#000000" size={14} />).toJSON()
  expect(tree).toMatchSnapshot()
})
