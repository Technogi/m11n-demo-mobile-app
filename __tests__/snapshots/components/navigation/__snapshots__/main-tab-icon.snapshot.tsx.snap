// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTabIcon renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderRadius": 17.5,
      "height": 35,
      "justifyContent": "center",
      "width": 35,
    }
  }
>
  <View
    style={
      {
        "marginBottom": 1.5714285714285714,
        "marginLeft": 1.5714285714285714,
      }
    }
  >
    <View
      style={
        {
          "marginBottom": 1.5714285714285714,
          "marginLeft": 1.5714285714285714,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#000000",
              "fontSize": 14,
            },
            undefined,
            {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ó°‹œ
      </Text>
    </View>
  </View>
</View>
`;
